File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> indexing = new Dictionary<int, int>(nums.Length); 
        
        for (int i = 0; i < nums.Length; i++) {
            int check = target - nums[i];
            
            if (indexing.TryGetValue(check, out int index)) {
                return new int[] { index, i };
            }
            
            indexing[nums[i]] = i;  // Update even if key exists (trusting no duplicates)
        }
        
        return null;    
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>indexing;
        vector<int> res;
        for(int i =0;i <nums.size();i++){
            int check = target - nums[i];

            if(indexing.find(check)!=indexing.end()){
                res.push_back(indexing[check]);
                res.push_back(i);
            }
            else{
                indexing[nums[i]] = i;
            }
        }
        return res;
    }
};